### PR TITLE
Create daemon service; Authenticate daemon service with GitHub Apps

### PR DESCRIPTION
**Pull request by MichaelDeng03**
- Add daemon server that can auth with github
- Configured webhook event handler that generates app installation token 
  - App installation token can be configured to authenticate GH api calls. 
- Add localhost access to services (in addition to tailscale)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,6 +40,9 @@ build-webapp-homepage:
 build-dind-builder:
 	docker compose -f docker/docker-compose.yml build dind-builder
 
+build-daemons:
+	docker compose --env-file .env.daemons -f docker/docker-compose.yml build daemons
```
Comment (MichaelDeng03) [inline Makefile:44]: I discovered how to add build args, instead of runtime args.

File: webapp/daemons/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# How to develop webhooks locally
+Webhooks require a URL to be able to send requests to. This is a problem when developing locally, as the webhook URL is not accessible from the internet. To solve this problem, I use ngrok to create a tunnel to my local machine. This allows me to expose my local server to the internet, and receive requests from the webhook.
```
Comment (kgang) [inline webapp/daemons/README.md:2]: Nice

File: webapp/daemons/server.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        "Authorization": f"Bearer {encoded_jwt}",
+        "X-GitHub-Version": "2022-11-28",
+    }
+    url = f'https://api.github.com/app/installations/{installation_id}/access_tokens'
+    token = RestApiTool.post(url=url, headers=headers)
+    token = token['token']
+    return token
+
+
+def verify_signature(payload_body, signature_header):
```
Comment (kgang) [inline webapp/daemons/server.py:82]: Interesting, was this recommended in the documentation or did you come up with this?

File: webapp/daemons/static/css/style.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
+body {
```
Comment (kgang) [inline webapp/daemons/static/css/style.css:1]: Did you make this?

**Review by kgang — APPROVED** [review]
LGTM

File: webapp/daemons/static/css/style.css (hunk 4 above)
Comment (MichaelDeng03) [reply webapp/daemons/static/css/style.css:1]: its copied from homepage

Comment (kgang) [reply webapp/daemons/static/css/style.css:1]: Ah [](edited-with-Axolo)

File: webapp/daemons/server.py (hunk 3 above)
Comment (MichaelDeng03) [reply webapp/daemons/server.py:82]: recommended in documentation
https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries

Comment (kgang) [reply webapp/daemons/server.py:82]: Do you mind adding this link to a comment with an explanation? [](edited-with-Axolo)

**Pull request merged**